Plutarch/Api/V1/Scripts.hs

```diff
@@ -64,3 +113,230 @@ deriving via
+
+-- | A compiled Validator, bundled with the Plutarch Term
+data Validator = Validator
+  { validatorScript :: Plutus.Script
+  , validatorTerm :: ClosedTerm PValidator
+  }
+
+-- | A compiled MintingPolicy, bundled with the Plutarch Term
+data MintingPolicy = MintingPolicy
+  { mintingPolicyScript :: Plutus.Script
+  , mintingPolicyTerm :: ClosedTerm PMintingPolicy
+  }
+
+-- | A compiled StakeValidator, bundled with the Plutarch Term
+data StakeValidator = StakeValidator
+  { stakeValidatorScript :: Plutus.Script
+  , stakeValidatorTerm :: ClosedTerm PStakeValidator
+  }
+
+-- | Compile a Validator
+mkValidator :: ClosedTerm PValidator -> Validator
+mkValidator term =
+  Validator (compile term) term
+
+-- | Compile a MintingPolicy
+mkMintingPolicy :: ClosedTerm PMintingPolicy -> MintingPolicy
+mkMintingPolicy term =
+  MintingPolicy (compile term) term
+
+-- | Compile a StakeValidator
+mkStakeValidator :: ClosedTerm PStakeValidator -> StakeValidator
+mkStakeValidator term =
+  StakeValidator (compile term) term
+
+-- | Serialise a Validator to a (Lazy) ByteString
+serialiseValidator :: Validator -> Lazy.ByteString
+serialiseValidator = serialiseScript . validatorScript
+
+-- | Serialise a StakeValidator to a (Lazy) ByteString
+serialiseStakeValidator :: StakeValidator -> Lazy.ByteString
+serialiseStakeValidator = serialiseScript . stakeValidatorScript
+
+-- | Serialise a MintingPolicy to a (Lazy) ByteString
+serialiseMintingPolicy :: MintingPolicy -> Lazy.ByteString
+serialiseMintingPolicy = serialiseScript . mintingPolicyScript
+
+-- | Hash a Script, with the correct prefix for Plutus V1
+scriptHash :: Plutus.Script -> Plutus.ScriptHash
+scriptHash = hashScriptWithPrefix "\x01"
+
+-- | Hash a Validator, with the correct prefix for Plutus V1
+validatorHash :: Validator -> Plutus.ValidatorHash
+validatorHash = coerce . scriptHash . validatorScript
+
+-- | Hash a MintingPolicy, with the correct prefix for Plutus V1
+mintingPolicySymbol :: MintingPolicy -> Plutus.CurrencySymbol
+mintingPolicySymbol = coerce . scriptHash . mintingPolicyScript
+
+-- | Hash a StakeValidator, with the correct prefix for Plutus V1
+stakeValidatorHash :: StakeValidator -> Plutus.StakeValidatorHash
+stakeValidatorHash = coerce . scriptHash . stakeValidatorScript
+
+{- |
+  Construct a `PValidator`, assuming the `PData` arguments are
+  of the correct form.
+
+  Shorthand for:
+  @pwrapValidatorWith punsafeAsData punsafeAsData punsafeAsData@
+-}
+pwrapValidatorAsData ::
+  ( forall s.
+    Term s (PAsData datum) ->
+    Term s (PAsData redeemer) ->
+    Term s (PAsData ctx) ->
+    Term s p
+  ) ->
+  ClosedTerm PValidator
+pwrapValidatorAsData = pwrapValidatorWith punsafeAsData punsafeAsData punsafeAsData
+
+{- |
+  Construct a `PValidator`, assuming the `PData` arguments are
+  of the correct form, wrapping them with `pfromData`.
+
+  Shorthand for:
+  @pwrapValidatorWith punsafeFromData punsafeFromData punsafeFromData@
+-}
+pwrapValidatorFromData ::
```

Comment:
These are *technically* unsafe, though it's perfectly valid to use them anyway in many cases...
Should they have `unsafe` in their names somehow? `punsafeWrapValidatorFromData` is pretty long...